service/reporter/reporter.go

```diff
@@ -342,6 +361,56 @@ func (s *reporter) collectTestLogs(folder string, fileNames chan string, tests [
 	return nil
 }
 
+
+// create an agency dump
+func (s *reporter) agencyDump(folder string, fileNames chan string, machines []cluster.Machine) error {
+
+	g := errgroup.Group{}
+	client := &http.Client{Timeout: time.Second * 5}
+
+	for _, m := range machines {
+		m := m // Used in nested func
+		filePrefix := fileNameFixer.Replace(m.ID())
+		if m.HasAgent() {
+			g.Go(func() error {
+				// Collect agent logs
+				if fileName, err := func() (string, error) {
+					f, err := os.Create(filepath.Join(folder, fmt.Sprintf("%s-agency-dump.json", filePrefix)))
+					if err != nil {
+						return "", maskAny(err)
+					}
+					defer f.Close()
+
+					configURL := m.AgentURL()
+					configURL.Path = "/_api/agency/state"
+					r, e := client.Get(configURL.String())
+
+					if e == nil && r != nil && r.StatusCode == 200 {
+						bodyBytes,_ := ioutil.ReadAll(r.Body)
+						ioutil.WriteFile(f.Name(), bodyBytes, 0644)
```

Comment:
You could use [`Copy`](https://golang.org/pkg/io/#Copy): `io.Copy(f, r.Body)`

Reply:
done